gemini/use-cases/document-rag-on-llamaindex/llamaindex_rag.ipynb

```diff
@@ -0,0 +1,1458 @@
+        "\n",
+        "    auth.authenticate_user()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4x10VEywwN3Z"
+      },
+      "source": [
+        "### Using LlamaIndex"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-2mDLUbdweJi"
+      },
+      "outputs": [],
+      "source": [
+        "from IPython.display import Image\n",
+        "\n",
+        "Image(filename=\"./LlamaRAG.png\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ybyxRabnmzlI"
+      },
+      "source": [
+        "### Investigate sample data\n",
+        "\n",
+        "Refer to document 04a02.pdf\n",
+        "\n",
+        "This document describes the importance of stable power grids in Japan, highlighting the recent failure of a generator step-up transformer at the Nakoso Power Station and the rapid restoration response undertaken to maintain power supply stability.\n",
+        "\n",
+        "We will use this pdf moving forward."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Zn2zpsSQmzlJ"
+      },
+      "source": [
+        "## Set Up: Vertex AI Index and Endpoint\n",
+        "\n",
+        "This section goes over how to set up the necessary cloud resources to run the RAG framework. You will be performing the following steps:\n",
+        "\n",
+        "1. Creating a Google Cloud bucket\n",
+        "2. Creating a Vertex AI Vector Search Index\n",
+        "3. Creating a Vertex AI Vector Search endpoint\n",
+        "4. Deploying a Vertex AI Vector Search endpoint\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "f7cD_tHsmzlJ"
+      },
+      "outputs": [],
+      "source": [
+        "def create_bucket_class_location(bucket_name):\n",
+        "    \"\"\"\n",
+        "    Create a new bucket in the US region with the coldline storage\n",
+        "    class.\n",
+        "    \"\"\"\n",
+        "\n",
+        "    storage_client = storage.Client()\n",
+        "\n",
+        "    # Searching for existing GCS bucket\n",
+        "    for bucket in storage_client.list_buckets():\n",
+        "        if bucket.name == bucket_name:\n",
+        "            print(f\"GCS Bucket {bucket_name} exists already in resource.\")\n",
+        "            return bucket\n",
+        "\n",
+        "    # Creating new bucket\n",
+        "    bucket = storage_client.bucket(bucket_name)\n",
+        "    bucket.storage_class = \"STANDARD\"\n",
+        "    new_bucket = storage_client.create_bucket(bucket, location=REGION)\n",
+        "\n",
+        "    print(\n",
+        "        \"Created bucket {} in {} with storage class {}\".format(\n",
+        "            new_bucket.name, new_bucket.location, new_bucket.storage_class\n",
+        "        )\n",
+        "    )\n",
```

Comment:
Generally prefer using f-strings instead of `.format()`

Reply:
Updated to f-strings.